tools/src/main/java/org/apache/kafka/tools/ToolsUtils.java

```diff
@@ -123,4 +125,21 @@ public static void validateBootstrapServer(String hostPort) throws IllegalArgume
             throw new IllegalArgumentException("Please provide valid host:port like host1:9091,host2:9092\n");
         }
     }
+
+    public static <T> List<T> duplicates(List<T> s) {
```

Comment:
Should the return type be `Set` instead of `List` as each entry in the returned value should only appear once?
If so this could be simplified into something like, so we only iterate the input list once:
```
Set<T> set = new HashSet<>();
Set<T> duplicates = new HashSet<>();
s.forEach(element -> {
    if (!set.add(element)) {
        duplicates.add(element);
    }
});
return duplicates;
```
Could we also add a short javadoc explaining the return value only contains each duplicate value once?